### PR TITLE
support src in --secret

### PR DESCRIPTION
**Pull request by vieux**
related to https://github.com/docker/docker/issues/28527

ping @dnephin @icecrime

### CODE REVIEW

**Comment by vieux** [comment]
ping @ehazlett

File: cli/command/service/opts.go [hunk 1]
```diff
@@ -182,7 +182,7 @@ func (o *SecretOpt) Set(value string) error {
 
 		value := parts[1]
 		switch key {
-		case "source":
+		case "source", "src":
```
Comment (thaJeztah) [inline cli/command/service/opts.go]: Looking at the UX for `docker secret ls`;

```bash
$ docker secret ls
ID                          NAME                    CREATED                                   UPDATED
mhv17xfe3gh6xc4rij5orpfds   secret.json             2016-10-27 23:25:43.909181089 +0000 UTC   2016-10-27 23:25:43.909181089 +0000 UTC
```

As a user I would expect `name` or `id` to be options for this, possibly `alias` for the name inside the container

/me continues painting bike shed

**Review by dnephin — APPROVED** [review]
LGTM

**Review by ehazlett — COMMENTED** [review]
LGTM

File: cli/command/service/opts.go (hunk 1 above)
Comment (thaJeztah) [inline cli/command/service/opts.go]: Yup, I see the issue. Difference with volumes is of course that `src` _does_ make sense if it's a bind-mount.

**Comment by runcom** [comment]
LGTM

**Pull request merged**